### ISSUE TITLE
Allow upload of video episodes

### ISSUE THREAD

**Issue by kookster** [PRX/cms.prx.org#306]
Several producers have video episodes, perhaps one or a couple, in their otherwise audio-only feeds

Feeder and dovetail handle this, but there is not a way in publish to upload or manage these files and episodes.

**Comment by kookster** [comment]
roman has some more of these coming too, in august

**Comment by kookster** [comment]
n.b. We need this for 99pi, which we will import to publish in August.

**Comment by kookster** [comment]
this definitely needs to get broken down, pls don't start on this until we do - 
here is my thinking on various specific changes that could be tix, perhaps the first 1 - 2 of publish would be appropriate for this sprint:

Publish:
- Pick a version template for a story when there are multiple (may work)
- Create multiple version templates for a series
- Set a version template media type
  - audio or video for now
  - For video, allow only a single file
- Allow upload by media type
  - if audio, allow only audio with validations
  - if video allow only video files, no validations

CMS:
- Allow audio versions to actually accept video
- Only apply audio validations to audio
- Add a media/content type to version templates
  - Add type to API

Dovetail:
- When feeder indicates template is a video, pass it through (should already do this based on content type)

**Comment by sandikbarr** [comment]
I _think_ what Publish does now with multiple version templates is makes you populate both when creating a story. Maybe it should be based on the media type selected for the story, and if you have multiple audio templates and choose audio for example, it makes you do all the audio versions. And there should only be one video template allowed with the single file, so you'd just have to do the one if your story type is video.

Not sure if CMS needs to be relaxed to allow publishing if there aren't audio versions on a story that has an audio template.

**Comment by cavis** [comment]
Yeah, Publish used to let you create multiple version-templates for a series.

But when you create a new story, it currently goes through _all_ of those version-templates, and creates that many audio-versions.  So yeah, some sort of dropdown/selector in the audio-versions that selects which one(s) to create.  And maybe some way to designate which audio-version-template(s) are "auto-created" on the backend.

**Comment by kookster** [comment]
@sandikbarr there will probably be multiple templates for audio for some series, like for the moth and on being, so we can't just choose the template by media type.

**Comment by kookster** [comment]
when a story is not in a series, we allow there to be no template, but for dovetail and other downstream services to work predictably, we need the structure of the show to match one we know about.

We're discussing an enhancement for "specials" or "promos" within a series that might not follow that structure, and maybe even be ad free, and such an exception may come in for those, but for an "episode", we need a template.

One option would be to work on this kind of "specials" feature, and that could be implemented with no structure/template, require only a single file, and some designation then to have it also not get validated.

That doesn't help for cases like the moth and on being that have 2 different kinds of episodes, but could help with an occasional 99pi video episode.

**Comment by sandikbarr** [comment]
@kookster right. What I was getting at is that currently when there are multiple templates, I believe that audio has to be provided to satisfy each template. We'll have to relax that in order to pick a single template and only provide media for that one.

Longer term, do we want to allow them to only pick a single template or pick one or more templates to create versions for a story?

**Comment by kookster** [comment]
@sandikbarr yeah, I'm fine with relaxing that "must have audio for all templates" requirement - that is a hold over from how V3 works, and not necessary. That should work for now.

Longer term?
What we have talked about is requiring versions based on the type of distribution - e.g. if this is a series distributed as a podcast, then, you need to have an audio version that satisfies being distributed as a podcast episode. If it is also distributed for podcast, you need a version that works for that - this could theoretically be the same version and files (e.g. 5 segments of wav/flac audio).

The requirement for what you need to upload is based on the required series distributions, not that you must provide audio for every possible template.

**Comment by kookster** [comment]
to sum up a bit - I could go either way - I think we'll need both capabilities eventually - 

1) "Special Episodes" - implement designating a story as a "special" type instead of an "episode", and then require single file, but otherwise perform no validations (this would still technically have an audio version, just no template for that version) - my spin on Sandi's suggestion

2) "Multiple Templates" - implement allowing a series to have multiple templates, and picking one at create time.

(1) is could work for the 99pi case, but (2) we need for really fixing TED, The Moth, and OnBeing, so that was my first thought

**Comment by kookster** [comment]
I also haven't thought enough about how "specials" might work, and there are ways they could work besides some of the assumptions above, so I'd also prefer to put that off until there is time to rationalize the new Apple designation of a "bonus" episode type with how specials might work - 

```
<itunes:episodeType> The type of episode you are submitting.
Where episodeType can be one of the following:
• full (default). Specify full when you are submitting the
complete content of a show.
• trailer. Specify trailer when you are submitting a short,
promotional piece of content that represents a preview of
a show.
• bonus. Specify bonus when you are submitting extra
content for a show (for example, behind the scenes
information or interviews with the cast).
```

**Comment by kookster** [comment]
I am breaking this down into other tickets, at least one in cms, and more in publish.

**Comment by sandikbarr** [comment]
@kookster Should we convert this to an epic since it's covered by all those tickets that are broken down now?

**Comment by cavis** [comment]
I'm going to close this ticket - see #345 for remaining work.
